tests/test_median_filter.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import numpy as np
+import torch
+
+from monai.networks.layers import MedianFilter
+
+
+class MedianFilterTestCase(unittest.TestCase):
+    def test_3d(self):
+        a = torch.ones(1, 1, 4, 3, 4)
+        g = MedianFilter(1).to(torch.device("cpu:0"))
+
+        expected = np.array(
```

Comment:
this is the failing test, right?

I think the expected outcome is different.

thinking about a 3x3x3 kernel, it has 27 positions, and
- on a corner of the image 19 of them (majority) are 0 due to padding
- on an edge of the image 15 of them (still majority) are 0 due to padding
- in the middle of a face 9 of them (not majority) are 0 due to padding

when the unpadded image consists of 1s, this means that when we apply the 3x3x3 median filter we expect:
- 0 at image corners
- 0 at image edges
- 1 in the middle of image faces
- 1 in the interior of the image volume